client/dashboard/profile-wizard/cart-modal.js

```diff
@@ -0,0 +1,134 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { Button, Modal } from '@wordpress/components';
+import { addQueryArgs } from '@wordpress/url';
+import { find } from 'lodash';
+import { decodeEntities } from '@wordpress/html-entities';
+
+/**
+ * WooCommerce dependencies
+ */
+import { getSetting } from '@woocommerce/wc-admin-settings';
+import { getNewPath } from '@woocommerce/navigation';
+import { List } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import withSelect from 'wc-api/with-select';
+import { getProductIdsForCart } from 'dashboard/utils';
+import sanitizeHTML from 'lib/sanitize-html';
+
+class CartModal extends Component {
+	componentDidMount() {
+		document.body.classList.add( 'woocommerce-admin-full-screen' );
+		document.body.classList.add( 'woocommerce-profile-wizard__body' );
+	}
+
+	componentWillUnmount() {
+		document.body.classList.remove( 'woocommerce-admin-full-screen' );
+		document.body.classList.remove( 'woocommerce-profile-wizard__body' );
+	}
+
+	redirectToCart() {
+		const { productIds } = this.props;
+		const backPath = getNewPath( {}, '/', {} );
+		const { connectNonce } = getSetting( 'onboarding', {} );
+
+		if ( ! productIds.length ) {
+			return;
+		}
+
+		const url = addQueryArgs( 'https://woocommerce.com/cart', {
+			'wccom-site': getSetting( 'siteUrl' ),
+			'wccom-woo-version': getSetting( 'wcVersion' ),
+			'wccom-replace-with': productIds.join( ',' ),
+			'wccom-connect-nonce': connectNonce,
+			'wccom-back': backPath,
+		} );
+		window.location = url;
+	}
+
+	renderProducts() {
+		const { productIds } = this.props;
+		const { productTypes = {}, themes = [] } = getSetting( 'onboarding', {} );
+		const listItems = [];
+
+		productIds.forEach( productId => {
+			const productInfo = find( productTypes, productType => {
+				return productType.product === productId;
+			} );
+
+			if ( productInfo ) {
+				listItems.push( {
+					title: productInfo.label,
+					content: productInfo.description,
+				} );
+			}
+
+			const themeInfo = find( themes, theme => {
+				return theme.id === productId;
+			} );
+
+			if ( themeInfo && themeInfo.slug !== 'Storefront' ) {
```

Comment:
Not opposed to separating Storefront and installing from the wp.org repo, but if we do that we may also want to check if this is the only product ID.  Otherwise, we should probably Storefront in this list since it will be added via wccom.

On a similar note, would we want to install any free themes from wp.org or explicitly set any that should bypass wccom?  If the former, we could check against price here to allow any themes that are free to be installed without visiting wccom.

Reply:
I think we should probably install any of the free themes from .org instead of WooCommerce.com. It saves multiple steps for the user if they didn't pick any other products, and it's less confusing than sending them to a cart with a $0.00 product.

It looks like the other free themes (like Deli) are in the .org repo.

What do you think about this being done in a follow-up? I think a couple pieces will need to be done. I can back out the `Storefront` check here and add a todo / log an issue.

Reply:
I think that's a good plan.  Outlined an issue for it [here](https://github.com/woocommerce/woocommerce-admin/issues/3455).